### ISSUE TITLE
Duplicate computers

### ISSUE THREAD

**Issue by ddurieux** [fusioninventory/fusioninventory-for-glpi#1684]
---

Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Redmine Issue: 3014, http://forge.fusioninventory.org/issues/3014
Original Date: 2015-09-01

---

GLPI version glpi-0.85.4 on Debian Jessie. <https://forge.indepnet.net/attachments/download/2020/glpi-0.85.4.tar.gz>
FusionInventory version: fusioninventory-for-glpi_0.85+1.2
FusionInventory Windows agent: fusioninventory-agent_windows-x86_2.3.16.exe (03/18/2015 04:32pm)

0) First time clean install of glpi with the fusionInventory server plugin. 
1) Installed Windows agent on a Windows XP. 
2) Shutdown Windows XP 
3) Shutdown GLPI server
4) Started GLPI server
5) After some time, started Windows XP

After some time, realized I now had duplicate Windows XP entries in GLPI despite the Windows XP is completely unchanged.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Date: 2015-09-01T22:45:02Z

---

I am attaching a csv export of the data (removed data for windows product key). I now got a second duplicate of the Windows XP system, though I do not believe I rebooted any of the systems.

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Date: 2015-09-02T02:15:17Z

---

There is only one agent for the windows machine registered in the table: glpi_plugin_fusioninventory_agents. 

2,0,1,'amadeus-2015-01-05-12-20-06 2015-08-31 22:07:38 {"INVENTORY":"v2.3.16"}',0,'amadeus-2015-01-05-12-20-06',4,12345678,'FusionInventory-Agent_v2.3.16',NULL,1,1,0,0,0,NULL

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Date: 2015-09-02T04:01:19Z

---

The "Computer" and "fusioninventorycomputer" arrays of The unserialized query:
SELECT serialized_inventory FROM `glpi_plugin_fusioninventory_inventorycomputercomputers` where id = x;
---------Extract id 2---------------------------------------------------

    [Computer] => Array
        (
            [users_id] => 0
            [name] => amadeus
            [operatingsystems_id] => 2
            [operatingsystemversions_id] => 2
            [uuid] => D23F7B05-3474-3559-BB08-B2B367EE4384
            [os_licenseid] => hidden
            [os_license_number] => hidden
            [domains_id] => 1
            [operatingsystemservicepacks_id] => 1
            [manufacturers_id] => 2
            [computermodels_id] => 2
            [serial] => 
            [computertypes_id] => 1
            [contact] => Admini@AMADEUS
            [is_dynamic] => 1
            [bmanufacturer] => Bochs
            [mserial] => 
            [states_id] => 0
        )

    [fusioninventorycomputer] => Array
        (
            [wincompany] => -
            [winowner] => hidden
            [operatingsystem_installationdate] => 2010-10-27 13:35:53
            [last_fusioninventory_update] => 2015-08-22 16:45:37
            [remote_addr] => 192.168.122.78
            [bios_assettag] => 
            [bios_date] => 2011-01-01
            [bios_version] => Bochs
            [bios_manufacturers_id] => 2
            [plugin_fusioninventory_computerarchs_id] => 2
        )

---------Extract id 3---------------------------------------------------
 
    [Computer] => Array
        (
            [users_id] => 0
            [name] => amadeus
            [operatingsystems_id] => 2
            [operatingsystemversions_id] => 2
            [uuid] => D23F7B05-3474-3559-BB08-B2B367EE4384
            [os_licenseid] => hidden
            [os_license_number] => hidden
            [domains_id] => 1
            [operatingsystemservicepacks_id] => 1
            [manufacturers_id] => 2
            [computermodels_id] => 2
            [serial] => 
            [computertypes_id] => 1
            [contact] => Admini@AMADEUS
            [is_dynamic] => 1
            [bmanufacturer] => Bochs
            [mserial] => 
            [states_id] => 0
        )

    [fusioninventorycomputer] => Array
        (
            [wincompany] => -
            [winowner] => hidden
            [operatingsystem_installationdate] => 2010-10-27 13:35:53
            [last_fusioninventory_update] => 2015-08-31 20:34:27
            [remote_addr] => 192.168.122.78
            [bios_assettag] => 
            [bios_date] => 2011-01-01
            [bios_version] => Bochs
            [bios_manufacturers_id] => 2
            [plugin_fusioninventory_computerarchs_id] => 2
        )

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Date: 2015-09-02T04:30:20Z

---

Might this be related with ./inc/lock.class.php line 190  related with 'glpi_plugin_fusioninventory_inventorycomputerblacklists' line 3, that indicate that indicate "Serial number", "ssn" cannot be null?

I should probably mention that my Windows XP is running as a KVM virtual guest system, so there is no serial number on the hardware itself as seen in the Computer array above. 

If this is the case, why does fusion inventory allow adding the equipment repetitively to GPLI, but block the data only when checking if it already is in the database?

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **David Durieux** (@ddurieux)
Original Date: 2015-09-02T20:16:10Z

---

It's not a bug, it's only a problem with your import rules not right by default for your case

**Comment by ddurieux** [comment]
---

Original Redmine Comment
Author Name: **Kenneth Andresen** (Kenneth Andresen)
Original Date: 2015-09-02T23:17:21Z

---

Thank you for your reference! I was trying to disable some of the block elements, but that I can see now was not the way to go. Your hint to the "import rules" made me find the page http://fusioninventory.org/documentation/documentation/fi4g/importrules.html 

I then found I could change and test(!) the rules within the FusionInventory plugin. A very nice feature not mentioned in the original install guide I had for GLPI pluss FusionInventory, but I in fact find very useful!

For myself it is ok not to consider this a bug - it has been resolved and I ok viewing it as a configuration issue. However, note that I did run this in a virtual environment for testing the solution on unimportant test machines before attempting to deploy to production, and the virtual test almost made me dismiss this solution and move on without ever knowing the rules were this flexible.

I will keep doing some more test cases, then hopefully start deploying this in the production environment.
